pkg/client/typed/discovery/discovery_client.go

```diff
@@ -209,75 +211,93 @@ func IsGroupDiscoveryFailedError(err error) bool {
 
 // serverPreferredResources returns the supported resources with the version preferred by the
 // server. If namespaced is true, only namespaced resources will be returned.
-func (d *DiscoveryClient) serverPreferredResources(namespaced bool) ([]schema.GroupVersionResource, error) {
+func (d *DiscoveryClient) serverPreferredResources(namespaced bool) ([]*unversioned.APIResourceList, error) {
 	// retry in case the groups supported by the server change after ServerGroup() returns.
 	const maxRetries = 2
```

Comment:
Long discussion here: https://github.com/kubernetes/kubernetes/pull/33234

Factored it out now. Much clearer.